test/revert.spec.ts

```diff
@@ -0,0 +1,5907 @@
+        getItemETH(parseEther("10"), parseEther("10"), seller.address),
+        getItemETH(parseEther("1"), parseEther("1"), reenterer.address),
+      ];
+
+      const { order, value } = await createOrder(
+        seller,
+        zone,
+        offer,
+        consideration,
+        0 // FULL_OPEN
+      );
+
+      // prepare the reentrant call on the reenterer
+      const callData = marketplaceContract.interface.encodeFunctionData(
+        "fulfillOrder",
+        [order, toKey(0)]
+      );
+      const tx = await reenterer.prepare(
+        marketplaceContract.address,
+        0,
+        callData
+      );
+      await tx.wait();
+
+      if (!process.env.REFERENCE) {
+        await expect(
+          marketplaceContract.connect(buyer).fulfillOrder(order, toKey(0), {
+            value,
+          })
+        ).to.be.revertedWith("NoReentrantCalls");
+      } else {
+        await expect(
+          marketplaceContract.connect(buyer).fulfillOrder(order, toKey(0), {
+            value,
+          })
+        ).to.be.reverted;
+      }
+    });
+  });
+
+  describe("ETH offer items", async () => {
+    let ethAmount: BigNumber;
+    const tokenAmount = minRandom(100);
+    let offer: OfferItem[];
+    let consideration: ConsiderationItem[];
+    let seller: Wallet;
+    let buyer: Wallet;
+
+    before(async () => {
+      ethAmount = parseEther("1");
+      seller = await getWalletWithEther();
+      buyer = await getWalletWithEther();
+      zone = new ethers.Wallet(randomHex(32), provider);
+      offer = [getItemETH(ethAmount, ethAmount)];
+      consideration = [getTestItem20(tokenAmount, tokenAmount, seller.address)];
+    });
+
+    it("fulfillOrder reverts if any offer item is ETH", async () => {
+      const { order, value } = await createOrder(
+        seller,
+        zone,
+        offer,
+        consideration,
+        0 // FULL_OPEN
+      );
+
+      await expect(
+        marketplaceContract
+          .connect(buyer)
+          .fulfillOrder(order, toKey(0), { value })
+      ).to.be.revertedWith("InvalidNativeOfferItem");
+    });
+
+    it("fulfillAdvancedOrder reverts if any offer item is ETH", async () => {
+      const { order } = await createOrder(
+        seller,
+        zone,
+        offer,
+        consideration,
+        0 // FULL_OPEN
+      );
+
+      await expect(
+        marketplaceContract
+          .connect(buyer)
+          .fulfillAdvancedOrder(order, [], toKey(0), buyer.address, {
+            value: ethAmount,
+          })
+      ).to.be.revertedWith("InvalidNativeOfferItem");
+    });
+
+    it("fulfillAvailableOrders reverts if any offer item is ETH", async () => {
+      const { order } = await createOrder(
+        seller,
+        zone,
+        offer,
+        consideration,
+        0 // FULL_OPEN
+      );
+
+      await expect(
+        marketplaceContract
+          .connect(buyer)
+          .fulfillAvailableOrders(
+            [order],
+            [[[0, 0]]] as any,
```

Comment:
nit: is there a more specific type we can use? Does `fulfillAvailableOrders` have typed args?

Reply:
it does, but it's expecting this generated type from `typechain-types`:
```typescript
export type FulfillmentComponentStruct = {
  orderIndex: BigNumberish;
  itemIndex: BigNumberish;
};
```
any thoughts how to handle so I can remove the `as any` ?